src/figuresClasses.ts

```diff
@@ -1,19 +1,113 @@
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shapes,
+  color: Color,
+  getArea(): number
+}
 
+function validator(key: string, value: number):void {
+  if (value <= 0) {
+    throw new Error(`Please enter valid ${key} value`);
+  }
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
+
+  color: Color;
+
+  a: number;
```

Comment:
In my opinnion it's better to write properties in constructor as `public a: number`, not to duplicate code